### PR TITLE
4x4 board

### PR DESCRIPTION
**Pull request by emashliles**
@gemcfadyen @jsuchy

### CODE REVIEW

File: src/com/TTT/Board.java [hunk 1]
```diff
@@ -7,20 +7,25 @@
...
 
-    public Board() {
-        spaces = new String[]{"1","2","3","4","5","6","7","8","9"};
+    public Board(int size) {
+        if(size == 4) {
+            spaces = new String[]{"1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16"};
+        }
+        else {
+            spaces = new String[]{"1","2","3","4","5","6","7","8","9"};
+        }
```
Comment (emashliles) [inline src/com/TTT/Board.java:16]: Still not happy with using an array of strings for this - it was the simplest thing at the time but I seem to have missed a more opportune point at which to refactor. I think I will probably refactor the use of X and O as strings first, to make this bit easier.

File: src/com/TTT/PlayerMarkers.java [hunk 2]
```diff
@@ -1,16 +1,22 @@
 package com.TTT;
 
 public enum PlayerMarkers {
-    X ("\u001B[31m " + "X" + " \u001B[0m"),
-    O("\u001B[34m " + "O" + " \u001B[0m");
+    X ("X", "\u001B[31m  X \u001B[0m"),
+    O ("O", "\u001B[34m  O \u001B[0m");
```
Comment (emashliles) [inline src/com/TTT/PlayerMarkers.java:6]: Looking at this, an enum may not have been the way to go. I played about with adding an empty space to this and it doesn't fit with the rest of it.

File: src/com/TTT/GameOptionsUI.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.TTT;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+public class GameOptionsUI {
```
Comment (emashliles) [inline src/com/TTT/GameOptionsUI.java:7]: In hindsight, GameOptions doesn't make sense as a class name and I was thinking ahead too much.

File: src/com/TTT/Board.java (hunk 1 above)
Comment (gemcfadyen) [reply src/com/TTT/Board.java:16]: You could initialise using a loop? That way if we introduced 5x5, you wouldnt have to make any more changes here.

File: src/com/TTT/BoardPrinter.java [hunk 4]
```diff
@@ -1,49 +1,56 @@
...
+        for (int rowIndex = 0; rowIndex < row.size(); rowIndex++) {
+            printSpace(board, row.getSpaceIndex(rowIndex));
+
+            if ((rowIndex + 1) % row.size() != 0)
+                out.print("|");
         }
     }
 
+    private String lineSeparator(int rowSize) {
+        int lineLength = (5 * rowSize) - 1;
```
Comment (gemcfadyen) [inline src/com/TTT/BoardPrinter.java:39]: what is the significance of the 5 here?

File: src/com/TTT/GameOptionsUI.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        this.in = in;
+        scanner = new Scanner(in);
+        scanner.useDelimiter("\n");
+    }
+
+    public int boardSize() {
+        out.print("Would you like a 3x3 or 4x4 board? Please enter 3 or 4: ");
+
+        String selection = getBoardSizeSelection();
+        while(!validSelection(selection) || ((Integer.parseInt(selection) != 3) && (Integer.parseInt(selection) != 4))) {
```
Comment (gemcfadyen) [inline src/com/TTT/GameOptionsUI.java:23]: You could wrap these three method calls into one so the while loop reads nicer `while(invalidInput(selection)`

**Review by gemcfadyen — COMMENTED** [review]
Noticed that the CI build is still broken.

**Pull request merged**